pingcap/tiflow
# changefeed (ticdc): Mask sensitive information in changefeed info

### What problem does this PR solve?


Issue Number: close #6362 

### What is changed and how it works?


### Check List 

#### Tests 


 - Manual test (add detailed scripts or steps below)
1. changefeed info in log
```shell
[2022/08/19 16:28:23.560 +08:00] [INFO] [etcd_worker.go:466] ["[etcd worker] ==========Update State to ETCD=========="]
[2022/08/19 16:28:23.560 +08:00] [INFO] [etcd_worker.go:481] ["[etcd worker] update changefeed info"] [key=/tidb/cdc/changefeed/info/test-1] [value="{\"upstream-id\":0,\"sink-uri\":\"mysql://root:xxxx@127.0.0.1:3306/?time-zone=\",\"opts\":{},\"create-time\":\"2022-08-19T16:13:58.502125+08:00\",\"start-ts\":435394140828598273,\"target-ts\":0,\"admin-job-type\":0,\"sort-engine\":\"unified\",\"sort-dir\":\"\",\"config\":{\"case-sensitive\":true,\"enable-old-value\":true,\"force-replicate\":false,\"check-gc-safe-point\":true,\"filter\":{\"rules\":[\"test1.t1\",\"test1.t2\",\"test1.t3\",\"test1.t4\"],\"ignore-txn-start-ts\":null},\"mounter\":{\"worker-num\":16},\"sink\":{\"dispatchers\":null,\"protocol\":\"\",\"column-selectors\":null,\"schema-registry\":\"\"},\"cyclic-replication\":{\"enable\":false,\"replica-id\":0,\"filter-replica-ids\":null,\"id-buckets\":0,\"sync-ddl\":false},\"consistent\":{\"level\":\"none\",\"max-log-size\":64,\"flush-interval\":1000,\"storage\":\"\"}},\"state\":\"normal\",\"error\":null,\"sync-point-enabled\":false,\"sync-point-interval\":600000000000,\"creator-version\":\"v6.0.0-master-dirty\"}"]
[2022/08/19 16:28:23.601 +08:00] [INFO] [etcd_worker.go:466] ["[etcd worker] ==========Update State to ETCD=========="]
```

2. changefeed info in cli
```shell
Create changefeed successfully!
ID: test-4
Info: {"upstream-id":0,"sink-uri":"mysql://root:xxxx@127.0.0.1:3306/?time-zone=","opts":{},"create-time":"2022-08-19T16:23:54.568248+08:00","start-ts":435394297067208706,"target-ts":0,"admin-job-type":0,"sort-engine":"unified","sort-dir":"","config":{"case-sensitive":true,"enable-old-value":true,"force-replicate":false,"check-gc-safe-point":true,"filter":{"rules":["test1.t1","test1.t2","test1.t3","test1.t4"],"ignore-txn-start-ts":null},"mounter":{"worker-num":16},"sink":{"dispatchers":null,"protocol":"","column-selectors":null,"schema-registry":""},"cyclic-replication":{"enable":false,"replica-id":0,"filter-replica-ids":null,"id-buckets":0,"sync-ddl":false},"consistent":{"level":"none","max-log-size":64,"flush-interval":1000,"storage":""}},"state":"normal","error":null,"sync-point-enabled":false,"sync-point-interval":600000000000,"creator-version":"v6.0.0-master-dirty"}
```

3. changefeed info in api
```shell
[2022/08/19 16:52:19.795 +08:00] [INFO] [open.go:319] ["Create changefeed successfully!"] [id=test6] [changefeed="{\"upstream-id\":0,\"sink-uri\":\"mysql://root:xxxx@127.0.0.1:3306/?time-zone=\",\"opts\":{},\"create-time\":\"2022-08-19T16:52:19.75472+08:00\",\"start-ts\":435394744030068738,\"target-ts\":0,\"admin-job-type\":0,\"sort-engine\":\"unified\",\"sort-dir\":\"\",\"config\":{\"case-sensitive\":true,\"enable-old-value\":true,\"force-replicate\":false,\"check-gc-safe-point\":true,\"filter\":{\"rules\":[\"*.*\"],\"ignore-txn-start-ts\":null},\"mounter\":{\"worker-num\":16},\"sink\":{\"dispatchers\":null,\"protocol\":\"\",\"column-selectors\":null,\"schema-registry\":\"\",\"transaction-atomicity\":\"table\"},\"cyclic-replication\":{\"enable\":false,\"replica-id\":0,\"filter-replica-ids\":null,\"id-buckets\":0,\"sync-ddl\":false},\"consistent\":{\"level\":\"none\",\"max-log-size\":64,\"flush-interval\":2000,\"storage\":\"\"}},\"state\":\"normal\",\"error\":null,\"sync-point-enabled\":false,\"sync-point-interval\":600000000000,\"creator-version\":\"v6.1.0-master-dirty\"}"]
[2022/08/19 16:52:19.795 +08:00] [INFO] [middleware.go:41] [/api/v1/changefeeds] [status=202] [method=POST] [path=/api/v1/changefeeds] [query=] [ip=127.0.0.1] [user-agent="apifox/1.0.0 (https://www.apifox.cn)"] [] [duration=523.349333ms]
```



#### Questions 

##### Will it cause performance regression or break compatibility?

##### Do you need to update user documentation, design documentation or monitoring documentation?

### Release note 

```release-note
None
```
